doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,201 @@
+configuration file. These settings are enabled by `CONFIG_NET_APP_SETTINGS`
+configuration option. This option is disabled by default. Note that for
+example the IPv6 options are only available if generic IPv6 support is enabled.
+
+* **CONFIG_NET_APP_MY_IPV6_ADDR** option sets a static IPv6 address to the
+  system. This is typically only useful in device testing as normally the
+  system should use SLAAC (IPv6 Stateless Address Auto Configuration) which
+  is enabled by default in the system. System can be configured to use
+  multiple IPv6 addresses, this is controlled by the
+  `CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT` configuration option.
+
+* **CONFIG_NET_APP_PEER_IPV6_ADDR** option tells what is the peer device
+  IPv6 address. This is only useful when testing client/server type
+  applications. This peer address is typically used as a parameter when
+  calling `net_app_connect()` function.
+
+* **CONFIG_NET_APP_MY_IPV4_ADDR** option sets a static IPv4 address to the
+  system. This is typically useful only in device testing as normally the
+  system should use DHCPv4 to discover the IPv4 address.
+
+* **CONFIG_NET_APP_PEER_IPV4_ADDR** option tells what is the peer device
+  IPv4 address. This is only useful when testing client/server type
+  applications. This peer address is typically used as a parameter when
+  connecting to other device.
+
+Next options are only available if IEEE 802.15.4 wireless network technology
+support is enabled.
+
+* **CONFIG_NET_APP_IEEE802154_DEV_NAME** tells the name of the IEEE 802.15.4
+  device.
+
+* **CONFIG_NET_APP_IEEE802154_PAN_ID** tells the used PAN identifier.
+  Note that the PAN id can be changed during runtime if needed.
+
+* **CONFIG_NET_APP_IEEE802154_CHANNEL** tells the used radio channel.
+  Note that the used channel can be changed during runtime if needed.
+
+* **CONFIG_NET_APP_IEEE802154_RADIO_TX_POWER** tells the initial radio
+  TX power level. The used TX power level can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_KEY** tells the initially used
+  security key. The used security key can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_KEY_MODE** tells the initially used
+  security key mode. The used security key mode can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_LEVEL** tells the initially used
+  security level. The used security level can be changed during runtime
+  if needed.
+
+Client / Server Applications
+############################
+
+The net-app API provides functions that enable the application to create
+a client / server type applications easily. If needed, the applications can
+have the communication secured by TLS (for TCP connections) or DTLS (for
+UDP connections) automatically.
+
+A simple **TCP server** application would consists of following net-app API
+function calls:
+
+* `net_app_init_tcp_server()` will configure used TCP port and local address.
```

Comment:
Perhaps s/will configure/to configure/ throughout this section.

Probably also s/used TCP port and local address/a local address and TCP port/